### PR TITLE
chore: remove my ppx and use the maintainer's

### PR DESCRIPTION
**Pull request by emjin**
Test plan:
See instructions in notion "How to add tracing" to get this running

### CODE REVIEW

File: libs/tracing/Tracing.mli [hunk 1]
```diff
@@ -6,22 +6,8 @@
 (* Functions to instrument the code *)
 (*****************************************************************************)
 
-val with_span :
```
Comment (aryx) [inline libs/tracing/Tracing.mli]: we might want to keep that function; you never know, sometimes the ppx is not appropriate.

File: src/core_scan/Core_scan.ml [hunk 2]
```diff
@@ -522,7 +522,7 @@ let sanity_check_invalid_patterns (res : Core_result.t) :
 (*****************************************************************************)
 
 (* for -rules *)
-let rules_from_rule_source (config : Core_scan_config.t) :
+let%trace rules_from_rule_source (config : Core_scan_config.t) :
```
Comment (aryx) [inline src/core_scan/Core_scan.ml:527]: This is really ugly. Is there a way to use attributes instead of those special let?

**Review by aryx — APPROVED** [review]
CI is failing

File: src/core_scan/Core_scan.ml (hunk 2 above)
Comment (aryx) [reply src/core_scan/Core_scan.ml:527]: For example using let%xxx will prevent to use simple semgrep patterns to find definitions
in OCaml; I'd rather use attributes.

Comment (emjin) [reply src/core_scan/Core_scan.ml:527]: This is how their ppx works. I don't think there's an alternative. https://github.com/c-cube/ocaml-trace

Comment (aryx) [reply src/core_scan/Core_scan.ml:527]: Can you ask them whether they could also support the [@@trace] syntax?

File: libs/tracing/Tracing.mli (hunk 1 above)
Comment (emjin) [reply libs/tracing/Tracing.mli]: Fair enough

File: src/core_scan/Core_scan.ml (hunk 2 above)
Comment (emjin) [reply src/core_scan/Core_scan.ml:527]: Hm, I'm not sure how they'd make it work with their syntax to add data to span, which I do find nice (see `handle_target_with_trace` for an example). I anticipate that we'll use it pretty often to tag with rule or file data that will be useful for filtering; it's the main reason I was determined to switch to their ppx.

Comment (nmote) [reply src/core_scan/Core_scan.ml:527]: I agree with @emjin -- this syntax doesn't offend me and I think it's nice to have one way of doing a particular thing. They've chosen this syntax and that is okay with me.

**Pull request merged**